### PR TITLE
fix(vscode-github): correctly obtain Git remote and parse it

### LINKED ISSUE #33: Bug: Not recognising GitHub remote with credencials
The extension seems to not recognising GitHub remote with credentials inside like `https://user:pass@github.com/user/repo`

Error at DevTools:

```
//Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:130 Not a github project?: Error: Not a github project?
    at execa.then.result (/Users/tino/.vscode-insiders/extensions/KnisterPeter.vscode-github-0.7.0/out/src/git.js:17:23)
    at process._tickCallback (internal/process/next_tick.js:103:7)e.onUnexpectedError @ 
```

### PR DESCRIPTION
**Pull request by tinovyatkin**
Correctly obtains Git remote and parses URL. Fixes #33

### CODE REVIEW

File: src/git.ts [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
-      const match = result.stdout.match(
-        /^remote\.origin\.url (?:git@github\.com:|https:\/\/github.com\/)(.*?)\/(.*?)(?:.git)?$/);
-      if (!match) {
-        throw new Error('Not a github project?');
-      }
-      return [match[1], match[2]];
-    });
+  // As we expect this function to throw on non-Github repos we can chain whatever calls and they will thrown on non-correct remotes
+  const remote = (await execa('git', 'config --local --get remote.origin.url'.split(' '), {cwd})).stdout.trim();
+  if (!remote.length) throw new Error('Git remote is empty!');
```
Comment (KnisterPeter) [inline src/git.ts]: Please always surround if statements with `{` and `}`.

File: src/git.ts [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
+
+  // Git protocol remotes, may be git@github or git://github, GITHUB domain name is not case-sensetive
+  if(remote.startsWith('git')) return remote.match(/^git(@|\:\/\/)github.com\/(.*?)\/(.*?)(?:.git)?$/i).slice(2, 4);
```
Comment (KnisterPeter) [inline src/git.ts]: See above. Please add curly braces here as well.

File: src/git.ts [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
+
+  // it must be http or https based remote
+  const { hostname, pathname } = parse(remote);
+  // Github.com domain name is not case-sensetive
+  if (!/^github\.com$/i.test(hostname)) throw new Error('Not a Github remote!');
```
Comment (KnisterPeter) [inline src/git.ts]: Please add curly braces here.

**Comment by KnisterPeter** [comment]
Besides my comments this looks very nice. Thank you!

**Comment by tinovyatkin** [comment]
I lack experience building with TS, so, it's plain ES6, not sure what I need to do for Travis :(

**Comment by KnisterPeter** [comment]
Thats the typescript compiler telling you that you do unsafe operations on the result of the regexp.match. That can be null which will result in an error (potentially overlooked by a developer).
I'll care about that. Don't mind.

Thanks for your help!

**Comment by KnisterPeter** [comment]
Closed with #38

**Pull request closed without merging**